src/1AC760.c

```diff
@@ -141,8 +145,652 @@ s32 calc_partner_test_enemy(void) {
 INCLUDE_ASM(s32, "1AC760", calc_partner_test_enemy);
 #endif
 
+s32 calc_partner_damage_enemy(void) {
+    BattleStatus* battleStatus = &gBattleStatus;
+    Actor* partner = battleStatus->partnerActor;
+    s32 currentTargetID = battleStatus->currentTargetID;
+    s32 currentTargetPartID = battleStatus->currentTargetPart;
+    s32 retVal;
+    s32 sp2C = FALSE;
+    s32 isFireDamage = FALSE;
+    s32 isElectricDamage = FALSE;
+    s32 isIceDamage = FALSE;
+    s32 tempBinary = FALSE;
+    s32 wasStatusInflicted = FALSE;
+    Actor* target;
+    ActorPart* targetPart;
+    ActorState* state;
+    Evt* evt;
+    s32 statusChanceOrDefense;
+    s32 damageDealt;
+    s32 dispatchEvent;
+
+    battleStatus->wasStatusInflicted = FALSE;
+    battleStatus->lastAttackDamage = FALSE;
+    battleStatus->attackerActorID = partner->actorID;
+    battleStatus->currentTargetID2 = battleStatus->currentTargetID;
+    battleStatus->currentTargetPart2 = battleStatus->currentTargetPart;
+    target = get_actor(currentTargetID);
+    state = &partner->state;
+
+    if (target == NULL) {
+        return 0;
+    }
+
+    targetPart = get_actor_part(target, currentTargetPartID);
+
+    if (targetPart == NULL) {
```

Comment:
This should probably instead be ASSERT(targetPart != NULL) but if I don't find anytihng else, I'll just merge it and fix it later